lib/taskana-core-test/src/test/java/acceptance/jobs/HistoryCleanupJobAccTest.java

```diff
@@ -0,0 +1,523 @@
+    }
+  }
+
+  @Nested
+  @TestInstance(Lifecycle.PER_CLASS)
+  class SimpleHistoryCleanupJobAllCompletedSameParentBusinessIsFalse
+      implements TaskanaConfigurationModifier {
+    @TaskanaInject TaskanaEngine taskanaEngine;
+    SimpleHistoryServiceImpl historyService;
+
+    @Override
+    public Builder modify(Builder builder) {
+      return builder.simpleHistoryCleanupJobAllCompletedSameParentBusiness(false);
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @BeforeAll
+    void setup() throws Exception {
+      historyService = new SimpleHistoryServiceImpl();
+      historyService.initialize(taskanaEngine);
+    }
+
+    @Test
+    @WithAccessId(user = "admin")
+    void should_CleanHistoryEventsUntilDate_When_SameParentBusinessFalse() throws Exception {
+      TaskHistoryEvent eventToBeCleaned =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId1",
+              TaskHistoryEventType.CREATED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned2 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId1",
+              TaskHistoryEventType.COMPLETED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned2.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned2.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned3 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId2",
+              TaskHistoryEventType.CREATED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned3.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned3.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned4 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId2",
+              TaskHistoryEventType.CANCELLED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned4.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned4.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned5 =
+          createTaskHistoryEvent(
+              "wbKey1",
+              "taskId3",
+              TaskHistoryEventType.CREATED.getName(),
+              "wbKey2",
+              "someUserId",
+              "someDetails");
+      eventToBeCleaned5.setCreated(Instant.now().minus(20, ChronoUnit.DAYS));
+      eventToBeCleaned5.setParentBusinessProcessId("sameParentId");
+
+      TaskHistoryEvent eventToBeCleaned6 =
```

Comment:
Can we reduce the number of events to 4?

Reply:
Yes we can